### ISSUE TITLE
Cron ran many times between 12am and 2am

### ISSUE THREAD

**Issue by jpepper07** [merencia/node-cron#35]
I have a cron job that I want to run every day at 4am. I thought I had the proper cron string for that, however it ran roughly 7200+ times. 

I have the following code in nodejs app.

```
var cron = new CronJob('* * 4 * * *', function(){
            console.log('Executing Cron: mongoDBCleanUp');
            var worker = require('./workers/mongoDBCleanUp');
            worker.execute();
        }, null, true, "America/New_York", null, false);
```

This is part of my logging via Papertrail.

```
Nov 01 01:59:53 pmx-scheduler app/worker.1:  Executing Cron: mongoDBCleanUp 
Nov 01 01:59:53 pmx-scheduler app/worker.1:  mongodb clean up running 
Nov 01 01:59:54 pmx-scheduler app/worker.1:  ready! 
Nov 01 01:59:54 pmx-scheduler app/worker.1:  Clean up for collection: uploads with query: {"uploadDate":{"$lte":"2016-10-31T08:59:54.170Z"}} 
Nov 01 01:59:54 pmx-scheduler app/worker.1:  Clean up for collection: exports with query: {"uploadDate":{"$lte":"2016-10-31T08:59:54.170Z"}} 
Nov 01 01:59:54 pmx-scheduler app/worker.1:  Clean up complete 
Nov 01 01:59:54 pmx-scheduler app/worker.1:  Executing Cron: mongoDBCleanUp 
Nov 01 01:59:54 pmx-scheduler app/worker.1:  mongodb clean up running 
Nov 01 01:59:55 pmx-scheduler app/worker.1:  ready! 
Nov 01 01:59:55 pmx-scheduler app/worker.1:  Clean up for collection: uploads with query: {"uploadDate":{"$lte":"2016-10-31T08:59:55.073Z"}} 
Nov 01 01:59:55 pmx-scheduler app/worker.1:  Clean up for collection: exports with query: {"uploadDate":{"$lte":"2016-10-31T08:59:55.073Z"}} 
Nov 01 01:59:55 pmx-scheduler app/worker.1:  Clean up complete 
Nov 01 01:59:55 pmx-scheduler app/worker.1:  Executing Cron: mongoDBCleanUp 
Nov 01 01:59:55 pmx-scheduler app/worker.1:  mongodb clean up running 
Nov 01 01:59:56 pmx-scheduler app/worker.1:  ready! 
Nov 01 01:59:56 pmx-scheduler app/worker.1:  Clean up for collection: uploads with query: {"uploadDate":{"$lte":"2016-10-31T08:59:56.262Z"}} 
Nov 01 01:59:56 pmx-scheduler app/worker.1:  Clean up for collection: exports with query: {"uploadDate":{"$lte":"2016-10-31T08:59:56.262Z"}} 
Nov 01 01:59:56 pmx-scheduler app/worker.1:  Clean up complete 
Nov 01 01:59:56 pmx-scheduler app/worker.1:  Executing Cron: mongoDBCleanUp 
Nov 01 01:59:56 pmx-scheduler app/worker.1:  mongodb clean up running 
Nov 01 01:59:57 pmx-scheduler app/worker.1:  ready! 
Nov 01 01:59:57 pmx-scheduler app/worker.1:  Clean up for collection: uploads with query: {"uploadDate":{"$lte":"2016-10-31T08:59:57.148Z"}} 
Nov 01 01:59:57 pmx-scheduler app/worker.1:  Clean up for collection: exports with query: {"uploadDate":{"$lte":"2016-10-31T08:59:57.148Z"}} 
Nov 01 01:59:57 pmx-scheduler app/worker.1:  Clean up complete 
Nov 01 01:59:57 pmx-scheduler app/worker.1:  Executing Cron: mongoDBCleanUp 
Nov 01 01:59:57 pmx-scheduler app/worker.1:  mongodb clean up running 
Nov 01 01:59:58 pmx-scheduler app/worker.1:  ready! 
Nov 01 01:59:58 pmx-scheduler app/worker.1:  Clean up for collection: uploads with query: {"uploadDate":{"$lte":"2016-10-31T08:59:58.200Z"}} 
Nov 01 01:59:58 pmx-scheduler app/worker.1:  Clean up for collection: exports with query: {"uploadDate":{"$lte":"2016-10-31T08:59:58.200Z"}} 
Nov 01 01:59:58 pmx-scheduler app/worker.1:  Clean up complete 
Nov 01 01:59:58 pmx-scheduler app/worker.1:  Executing Cron: mongoDBCleanUp 
Nov 01 01:59:58 pmx-scheduler app/worker.1:  mongodb clean up running 
Nov 01 01:59:59 pmx-scheduler app/worker.1:  ready! 
Nov 01 01:59:59 pmx-scheduler app/worker.1:  Clean up for collection: uploads with query: {"uploadDate":{"$lte":"2016-10-31T08:59:59.115Z"}} 
Nov 01 01:59:59 pmx-scheduler app/worker.1:  Clean up for collection: exports with query: {"uploadDate":{"$lte":"2016-10-31T08:59:59.115Z"}} 
Nov 01 01:59:59 pmx-scheduler app/worker.1:  Clean up complete 
```

**Comment by cwilsontv** [comment]
Your cron string isn't correct. You have 6 values and cron uses 5. I'm not sure how node-cron handles that, but obviously it's causing some unintended behavior.

Additionally, * is like a wild card for "all". So in first spot (the minute spot) it's like saying "every minute" so that might be part of the issue.

What you probably want is `0 4 * * *` which is 0 minutes, 4 hours, all other times. In other words "At 4:00 every day, every month, every day of the week".

Check out this site for help, I used them many times: https://crontab.guru/#0_4_*_*_*

**Comment by merencia** [comment]
Thanks for your answer @cwilsontv!

@jpepper07 
On node-cron the second is option an by default it's setted to 0. 
In other words this: `0 4 * * *` is equals to `0 0 4 * * *`.

When you use this expression `* * 4 * * *` you're asking to node-cron:
`Please, run this task every second of every minute of hour 4`.

Besides I think you're using another library, because `node-cron` doesn't support timezone. 

Check in your `package.json` if you are using [node-cron](https://www.npmjs.com/package/node-cron) or some other package like [cron](https://www.npmjs.com/package/cron), [node-schedule](https://www.npmjs.com/package/node-schedule) or another one.

**Comment by jpepper07** [comment]
@cwilsontv Thanks so much for the help and detail explanation. It all makes more sense now. I really appreciate the crontab.guru site. I was looking for something like this before.

@merencia Sorry, you are correct. I thought I was still configured using your node-cron. I started with it but I had trouble getting something like * * * * * working. So then I switched to cron. I am sure it was user error then.

Thanks All!
